modules/tensor_mechanics/include/kernels/ADWeakPlaneStress.h

```diff
@@ -0,0 +1,47 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ADKernelValue.h"
+
+#define usingWeakPlaneStressMembers                                                                \
+  usingKernelValueMembers;                                                                         \
+  using ADWeakPlaneStress<compute_stage>::_base_name;                                              \
+  using ADWeakPlaneStress<compute_stage>::_stress;                                                 \
+  using ADWeakPlaneStress<compute_stage>::_direction;
+
+// Forward Declarations
+template <ComputeStage>
+class ADWeakPlaneStress;
+
+declareADValidParams(ADWeakPlaneStress);
+
+/**
+ * ADWeakPlaneStress is the automatic differentiation version of WeakPlaneStress
+ */
+template <ComputeStage compute_stage>
+class ADWeakPlaneStress : public ADKernelValue<compute_stage>
+{
+public:
+  static InputParameters validParams();
+
+  ADWeakPlaneStress(const InputParameters & parameters);
+
+protected:
+  ADReal precomputeQpResidual() override;
+
+  const std::string _base_name;
```

Comment:
All member variables and methods (except for those marked override) need doxygen comments. The `ADComputeStrainIncrementBasedStress.h` file that you added has those (probably because the version it is based on already had them). If you could please add similar comments to the other header files you added, that would be great. A lot of our older code doesn't have those, but we're requiring them for any new code.

As you're doing this, if you could go back and add the same comments to the non-AD versions of these classes (since they will be essentially the same), that would be extra helpful.

Alternatively, I'd be happy to add this to the existing files, and you could just copy that over if you prefer. Let me know.

Reply:
I can add docs in the entire TM module for those members I touched in this PR.